### DISCUSSION TITLE
Auto redirect to app if user is logged on in Zitadel?

### DISCUSSION THREAD

**Discussion by helmut72** [zitadel/zitadel#4636] [Q&A] [answered]
Is there an option to enable auto redirect back to the app if an user is logged on in Zitadel?

For trying out Zitadel it's great to choose one my test users or the admin user. From work (Azure) and also from home with Keycloak or Authelia I'm used to be auto redirect back into the app when I'm logged on. IMHO it's unnecessary or do I oversee anything?

Thank you.

**Comment by hifabienne** [comment]
Hi @helmut72 

If you are sending an authorization request from your application you will have to send a redirect url, that is registered on your client After login we will automatically redirect you to the app.

If we do not have an authorization request we will redirect you to the default redirect url you have registered in the settings of your instance (login behaviour), if nothing is set the default redirect url points to the ZITADEL administration console.

**Comment by helmut72** [comment]
Hi @hifabienne 

thank you for your answer. Redirect url is setup and works. If a user opens app1.example.com, choose 'login with IDP', he will be redirected to Zitadel, type his password and will be redirected back to app1.
If a user then opens app2.example.com, choose 'login with IDP', he will be still redirected to Zitadels login page and must choose the username again:

<img width="424" alt="Bildschirm­foto 2022-10-31 um 09 00 58" src="https://user-images.githubusercontent.com/1388751/198960620-1d24819d-e3a2-43e3-b995-f70c8b9689d6.png">

Other systems (Azure, Keycloak, Authelia) do this automatically. A user just see some redirects in the browser url bar. This is of course a lot faster than rendering the login page and choose the username. That's why I ask if this behavior can be also configured in Zitadel as an option?

For me (as an admin) the current behavior is great for testing. From perspective of the user the login process takes of course a lot more time.

**Comment by hifabienne** [comment]
Hi @helmut72 

Actually I thought that this should already work. If you have only one user to select it should be chosen automatically. We will have a look into this. 
In the meantime, if you send the login_hint of your account in the authentication request, the user is also directly selected.

**Comment by helmut72** [comment]
Hi @hifabienne

> Actually I thought that this should already work. If you have only one user to select it should be chosen automatically.

That's why I have asked, if there is an option where I can enable this behavior? Maybe I haved missed something.

**Comment by hifabienne** [comment, marked as answer]
I have created an issue for this: https://github.com/zitadel/zitadel/issues/4841
But it will probably take a moment, since this is a breaking change

**Reply by helmut72** [reply]
Thank you 👍
